Hide unnecessarily exported symbols

Chibi libraries (accidentally) make some internal symbols public.
We don't want this and need to hide them.

"_huff_tab*" symbols are trickier in that they are used in both
sexp.c (locally) and in SRFI 95 (qsort.c), which expect them
to be exported when built as static library. That's why we only
mark them with __attribute__((visibility("hidden"))) so that
they are hidden from shared library exports, but are visible
when linking static libraries.

All public symbols are prefixed with "sexp_*".

--- a/include/chibi/sexp-hufftabs.c
+++ b/include/chibi/sexp-hufftabs.c
@@ -1,5 +1,7 @@
 /* auto-generated by ./symhuff.scm on Sun Feb 25 23:12:52 2007 */
 
+#pragma GCC visibility push(hidden)
+
 char _huff_tab21[] = {
   '\x01',  '\x00',  '\x03',  '\x00',  '\x02',  '\x00',  '\x04',  '\x00',
 };
@@ -90,3 +92,4 @@
      'e',     't',
 };
 
+#pragma GCC visibility pop
--- a/include/chibi/sexp-hufftabs.h
+++ b/include/chibi/sexp-hufftabs.h
@@ -1,5 +1,7 @@
 /* auto-generated by ./symhuff.scm on Sun Feb 25 23:12:52 2007 */
 
+#pragma GCC visibility push(hidden)
+
 char _huff_tab21[] = {
   '\x01',  '\x00',  '\x03',  '\x00',  '\x02',  '\x00',  '\x04',  '\x00',
 };
@@ -90,3 +92,4 @@
      'e',     't',
 };
 
+#pragma GCC visibility pop
